### PR TITLE
[CI] Add ability to rename benchmarking CI run results

### PR DESCRIPTION
**Pull request by ianayl**
- Adds ability to rename results of benchmarking CI runs
- Adds sanitization to all text inputs

### CODE REVIEW

**Comment by ianayl** [comment]
Test runs for all edge cases:
- unnamed: https://github.com/intel/llvm/actions/runs/16886371047
- named: https://github.com/intel/llvm/actions/runs/16893162766
- PR runs: https://github.com/intel/llvm/actions/runs/16915984413
- commit runs: https://github.com/intel/llvm/actions/runs/16917970419
- commit run, but overriding name: https://github.com/intel/llvm/actions/runs/16942083390/job/48013545392
- bad names (should fail): https://github.com/intel/llvm/actions/runs/16941726209

**Comment by ianayl** [comment]
@intel/dpcpp-devops-reviewers May I get a review? Thanks!

File: .github/workflows/sycl-ur-perf-benchmarking.yml [hunk 1]
```diff
@@ -90,16 +106,63 @@ on:
...
+      SAVE_NAME: ${{ inputs.save_name }}
+    outputs:
+      benchmark_save_name: ${{ steps.sanitize.outputs.benchmark_save_name }}
+      build_ref: ${{ steps.sanitize.outputs.build_ref }}
+    steps:
+      - id: sanitize
+        run: |
+          # Validate user inputs:
+          check_nonempty() {
+            # usage: check_if_nonempty <var> <regex to check var against> <err message>
```
Comment (uditagarwal97) [inline .github/workflows/sycl-ur-perf-benchmarking.yml]:
```suggestion
          # usage: check_if_nonempty <var> <regex to check var against> <err message>
          check_nonempty() {
```

File: .github/workflows/sycl-ur-perf-benchmarking.yml [hunk 2]
```diff
@@ -90,16 +106,63 @@ on:
...
+          check_nonempty "$PR_NO" '^[0-9]+$' "Bad PR number."
+          check_nonempty "$SAVE_NAME" '^[A-Za-z][A-Za-z0-9_-]+$' "Bad save name."
+
+          BENCHMARK_SAVE_NAME=""
+          BUILD_REF="${{ github.ref }}"
+          if [ -n "$SAVE_NAME" ]; then
+            BENCHMARK_SAVE_NAME="$(echo "$SAVE_NAME" | tr -cd 'A-Za-z0-9_-')"
+          fi;
+          if [ -n "$COMMIT_HASH" ]; then
+            commit_cleaned="$(echo "$COMMIT_HASH" | tr -cd 'a-z0-9')"
```
Comment (uditagarwal97) [inline .github/workflows/sycl-ur-perf-benchmarking.yml]: What does `tr -cd 'a-z0-9'` do? What does it mean to clean the commit hash? - because few lines above we are anyway checking if commit hash is valid or not.

Comment (ianayl) [reply .github/workflows/sycl-ur-perf-benchmarking.yml]: `-cd` is complement + delete: The command deletes all characters that isn't in `a-z0-9`.

This is definitely redundant, but I recall PSE's having a policy against ever using raw input from the user directly, especially if it has a potential of being later used in a command. However, if I just use `tr`, users wouldn't get an error if they inputted invalid input. So, the first few lines that check each input against a regex were mostly to notify the user incase they put in anything invalid.

I figured the time spent doing the extra regex check above the `tr` command (and vice versa) would be negligible, thus went for this instead.

Comment (uditagarwal97) [reply .github/workflows/sycl-ur-perf-benchmarking.yml]: TBH, I think we can just remove this. If user enters an invalid commit id or PR number, we'll just throw an error message and exit

Comment (ianayl) [reply .github/workflows/sycl-ur-perf-benchmarking.yml]: Removed

**Comment by ianayl** [comment]
@intel/llvm-gatekeepers PR ready for merge, thanks!

This PR only concerns CI changes, thus E2E failures are unrelated. Trackers for CI failures:
- Linux failures on HIP: https://github.com/intel/llvm/issues/17194
  - Interesting case: test output says "Test passed" but the test seems to crash last-minute
- Windows failures on Gen12: https://github.com/intel/llvm/issues/19354
